tests/test_flow/test_protocols.py

```diff
@@ -196,6 +199,100 @@ def df():
     pdt.assert_frame_equal(builder.build().get('df'), df_value)
 
 
+def test_simple_dask_dataframe(builder):
+    df_value = df_from_csv_str('''
+    color,number
+    red,1
+    blue,2
+    green,3
+    ''')
+    dask_df = dd.from_pandas(df_value, npartitions=1)
+
+    @builder
+    @bn.protocol.dask
+    @count_calls
+    def df():
+        return dask_df
+
+    assert equal_frame_and_index_content(builder.build().get('df').compute(), dask_df.compute())
+    assert equal_frame_and_index_content(builder.build().get('df').compute(), dask_df.compute())
+    assert df.times_called() == 1
+
+
+def test_multiple_partitions_dask_dataframe(builder):
+    df_value = df_from_csv_str('''
+    color,number
+    red,1
+    blue,2
+    green,3
+    ''')
+    dask_df = dd.from_pandas(df_value, npartitions=3)
+
+    @builder
+    @bn.protocol.dask
+    @count_calls
+    def df():
+        return dask_df
+
+    assert equal_frame_and_index_content(builder.build().get('df').compute(), dask_df.compute())
+    assert equal_frame_and_index_content(builder.build().get('df').compute(), dask_df.compute())
+    assert df.times_called() == 1
+
+
+def test_typed_dask_dataframe(builder):
+    df_value = pd.DataFrame()
+    df_value['int'] = [1, 2, 3]
+    df_value['float'] = [1.0, 1.5, float('nan')]
+    df_value['str'] = ['red', 'blue', None]
+    df_value['time'] = pd.to_datetime([
+        '2011-02-07',
+        '2011-03-17',
+        '2011-04-27',
+    ])
+    dask_df = dd.from_pandas(df_value, npartitions=1)
+
+    @builder
+    @bn.protocol.dask
+    def df():
+        return dask_df
+
+    assert equal_frame_and_index_content(builder.build().get('df').compute(), dask_df.compute())
+    assert builder.build().get('df').compute().dtypes.to_dict() ==\
+        dask_df.compute().dtypes.to_dict()
+
+
+@pytest.mark.skipif(sys.version_info.major == 2, reason="requires python 3")
```

Comment:
nit: should we use `six.PY3` here too?

Reply:
Definitely (assuming you mean `six.PY2`)!